### PR TITLE
Status DSL

### PR DESCRIPTION
**Pull request by manuelpuyol**
Extracted our current `self.status` logic to a DSL to make the components cleaner

### CODE REVIEW

**Review by joelhawksley — APPROVED** [review]
Nice! Love the DSL ❤️

**Review by keithamus — APPROVED** [review]
This looks like a great improvement :+1:

File: app/lib/primer/status/dsl.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      }.freeze
+
+      class UnknownStatusError < StandardError; end
+
+      included do
+        class_attribute :component_status, instance_writer: false, default: STATUSES[:alpha]
+      end
+
+      class_methods do
+        def status(status = nil)
```
Comment (srt32) [inline app/lib/primer/status/dsl.rb:33]: Is it typical to have this method be both the writer and reader (depending on if the arg is passed)?

Comment (manuelpuyol) [reply app/lib/primer/status/dsl.rb:33]: I borrowed this idea from the `view_component` framework. They do the same for SlotsV2 where `.slot_name` returns the slot and `.slot_name(xyz)` sets the slot

I thought it was a good way to not break the read API so I didn't have to change tests and dumps

**Pull request merged**